components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: appstudio-monitoring
+spec: {}
+---
+# Deploy Monitoring Stack
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: appstudio-monitoring-ms
+  namespace: appstudio-monitoring
+spec:
+  # Used to select the ServiceMonitor in the appstudio-monitoring namespace
+  # NOTE: there isn't a need for namespaceSelector
+  resourceSelector:
+    matchLabels:
+      monitoring.rhobs/stack: appstudio-monitoring-ms
+  logLevel: info # use debug for verbose logs
+  retention: 3h
+  prometheusConfig:
+    replicas: 2  # ensures that at least one prometheus is running during upgrade
+  alertmanagerConfig:
+    disabled: true
+  resources: # ensure that you provide sufficient amount of resources
+    requests:
+      cpu: 500m
+      memory: 1Gi
+---
+# Grant permission to Federate In-Cluster Prometheus
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: appstudio-monitoring-ms-view
+  labels:
+    kubernetes.io/part-of: appstudio-monitoring-ms
+    monitoring.rhobs/stack: appstudio-monitoring-ms
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+subjects:
+- kind: ServiceAccount
+  # ServiceAccount used in the prometheus deployed by ObO.
+  # SA name follows <monitoring stack name>-prometheus nomenclature
+  name: appstudio-monitoring-ms-prometheus
+  namespace: appstudio-monitoring
+---
+# Create ServiceMonitor for Federation
+apiVersion: monitoring.rhobs/v1
+kind: ServiceMonitor
+metadata:
+  name: appstudio-monitoring-smon
+  namespace: appstudio-monitoring
+  labels:
+    kubernetes.io/part-of: appstudio-monitoring-ms
+    monitoring.rhobs/stack: appstudio-monitoring-ms
+spec:
+  selector: # use the prometheus service to create a "dummy" target.
+    matchLabels:
+      app.kubernetes.io/managed-by: observability-operator
+      app.kubernetes.io/name: appstudio-monitoring-ms-prometheus
+  endpoints:
+  - params:
+      'match[]': # scrape only required metrics from in-cluster prometheus
+        - '{__name__="container_network_transmit_bytes_total"}'
```

Comment:
For now lets skip the following namespaces:
`openshift-*`
`kube-*`
`*-tenant`

Reply:
for all metrics

Reply:
done